### DISCUSSION TITLE
Multiple versions in App-Catalog after deploying the package

### DISCUSSION THREAD

**Discussion by RobertHoegner** [microsoft-search/pnp-modern-search#2051] [Q&A] [answered]
I have been using the WebParts for quite a while and wanted to upgrade to latest solution today. Unfortunately after deploying the package I ended up with two Versions inside my AppCatalog. I instead expected it to replace the existing. Maybe I did something wrong when deploying the app to the catalog. 
Would highly appreciate any suggestion on how to fix. Thank you very much in advance.

Screenshot of app-catalog:
![image](https://user-images.githubusercontent.com/49672998/165710905-be93382f-f808-43e0-bdd6-19ad1c0d9837.png)

Update: I can in fact use the new WebParts but how do I get rid of the old ones still showing up when I edit a page to place webpart:
![image](https://user-images.githubusercontent.com/49672998/165739446-621e2156-e0ca-44a0-aacd-450ea58c1dc0.png)

and inside the site contents of the site collection the app still shows the old version:
![image](https://user-images.githubusercontent.com/49672998/165739557-f99a1221-6c94-4fe4-8f20-0f087ff04ddc.png)

**Comment by wobba** [comment, marked as answer]
Version 3 and 4 are two independent major versions of the web parts and they can run side by side. If you want to remove one, then remove the app from the catalog and remove references to the web parts from pages where they are used.

There is no automatic upgrade from v3 to v4.

**Reply by RobertHoegner** [reply]
got it & love the web parts! thank you so much
